docs/howto/filesystem-management/data-transfer.md

```diff
@@ -0,0 +1,138 @@
+(migrate-data)=
+# Transfer data across filestores
+
+This documentation covers how to transfer data between filestores.
+
+This process should be repeated for as many hubs as there are on the cluster, remembering to update the value of `$HUB_NAME`.
+
+## The initial copy process
+
+```bash
+export CLUSTER_NAME=<cluster_name>
+export HUB_NAME=<hub_name>
+```
+
+1. **Create a pod on the cluster and mount the source and destination filestores.**
+
+   We can use the following deployer command to create a pod in the cluster with the two filestores mounted.
+   The current default filestore will be mounted automatically and we use `--extra-nfs-*` flags to mount the second filestore.
+
+   ```bash
+   deployer exec root-homes $CLUSTER_NAME $HUB_NAME \
+     --extra-nfs-server=$SERVER_IP \
+     --extra-nfs-base-path="/" \
+     --extra-nfs-mount-path="dest-fs" \
+     --persist
+   ```
+
+   - `$SERVER_IP` can be found either through the relevant Cloud provider console, or by running `kubectl --namespace $HUB_NAME get svc` if the second filestore is running `jupyterhub-home-nfs`.
+   - The `--persist` flag will prevent the pod from terminating when you exit it, so you can leave the transfer process running.
+
+1. **Install some tools into the pod.**
+
+   We'll need a few extra tools for this transfer so let's install them.
+
+   ```bash
+   apt-get update && \
+   apt-get install -y rsync parallel screen
+   ```
+
+   - `rsync` is what will actually perform the copy
+   - `parallel` will help speed up the process by parallelising it
+   - `screen` will help the process continue to live in the pod and be protected from network disruptions that would normally kill it
+
+1. **Start a screen session and begin the initial copy process.**
+
+   ```bash
+   ls /root-homes/${HUB_NAME}/ | parallel -j4 rsync -ah --progress /root-homes/${HUB_NAME}/{}/ /dest-fs/${HUB_NAME}/{}/
+   ```
+
+   ```{admonition} Monitoring tips
+   :class: tip
+
+   Start with `-j4`, monitor for an hour or so, and increase the number of threads until you reach high CPU utilisation (low `idle`, high `iowait` from the `top` command).
+   ```
+
+   ```{admonition} screen tips
+   :class: tip
+
+   To disconnect your `screen` session, you can input {kbd}`Ctrl` + {kbd}`A`, then {kbd}`D` (for "detach").
+
+   To reconnect to a running `screen` session, run `screen -r`.
+
+   Once you have finished with your `screen` session, you can kill it by inputting {kbd}`Ctrl` + {kbd}`A`, then {kbd}`K` and confirming.
+   ```
+
+   Once you have detached from `screen`, can now `exit` the pod and let the copy run.
+
+(migrate-data:reattach-pod)=
+## Reattaching to the data transfer pod
+
+You can regain access to the pod created for the data transfer using:
+
+```bash
+kubectl --namespace $HUB_NAME attach -i ${CLUSTER_NAME}-root-home-shell
```

Comment:
Would `attach` here cause problems when the user `exit`s, if it's pid 1?

Reply:
Good question that I'm not sure of the answer to.

I think `attach` would cause problems if it's PID 1, but we've taken steps to avoid getting assigned PID 1.

`attach` has the benefit of giving us the same process back. Does `exec` do the same if we do _not_ provide the `-t` flag? E.g. only `exec -i foo bar`

Reply:
I *think* `attach` will always give you the process the pod was opened with, which I think will be pid 1. This is how it gives us the same process back each time - by always giving us the same process that was used to start the pod. 

The easiest way to verify is to run `echo $$` ([some other similar vars](https://ryanstutorials.net/bash-scripting-tutorial/bash-variables.php#others)) (this prints the current pid) after `attach`, and see if  you get `1`, and repeat the process for `exec`. This should help you make the determination for wether we should use attach or exec here.

exec will *always* start a new process, regardless of whether `-i` is passed or not. So `screen` is required when using exec to get back to old terminal

Reply:
fixed in https://github.com/2i2c-org/infrastructure/pull/5284/commits/e892e82fb5f55dc0c77feed51a36a06624d7035d